Updating the examples

Poolitzer opened the pull request:
<!--
Hey! You're PRing? Cool! Please have a look at the below checklist. It's here to help both you and the maintainers to remember some aspects. Make sure to check out our contribution guide (https://github.com/python-telegram-bot/python-telegram-bot/blob/master/.github/CONTRIBUTING.rst).
-->

### Checklist for PRs

- [ ] Added `.. versionadded:: version`, `.. versionchanged:: version` or `.. deprecated:: version` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [ ] Created new or adapted existing unit tests
- [ ] Added myself alphabetically to `AUTHORS.rst` (optional)
- [ ] Added new classes & modules to the docs


### If the PR contains API changes (otherwise, you can delete this passage)

* New classes:
    - [ ] Added `self._id_attrs` and corresponding documentation
    - [ ] `__init__` accepts `**_kwargs`
    
* Added new shortcuts:
    - [ ] In `Chat` & `User` for all methods that accept `chat/user_id`
    - [ ] In `Message` for all methods that accept `chat_id` and `message_id`
    - [ ] For new `Message` shortcuts: Added `quote` argument if methods accepts `reply_to_message_id`
    - [ ] In `CallbackQuery` for all methods that accept either `chat_id` and `message_id` or `inline_message_id`
    
* If relevant:

    - [ ] Added new constants at `telegram.constants` and shortcuts to them as class variables
    - [ ] Added new handlers for new update types
    - [ ] Added new filters for new message (sub)types
    - [ ] Added or updated documentation for the changed class(es) and/or method(s)
    - [ ] Updated the Bot API version number in all places: `README.rst` and `README_RAW.rst` (including the badge), as well as `telegram.constants.BOT_API_VERSION`
    - [ ] Added logic for arbitrary callback data in `tg.ext.Bot` for new methods that either accept a `reply_markup` in some form or have a return type that is/contains `telegram.Message`

Bibo-Joshi reviewed (changes requested):
Thanks for the PR! changes themself look good except for the pre-commit errors. would you mind adding this section back to the pre-commit config?

https://github.com/python-telegram-bot/python-telegram-bot/blob/d117d21fffbb844b565f6cb9979c1c6afe601bc1/.pre-commit-config.yaml#L48-L59

I had removed it temporarily while working on the asyncio changes …

Poolitzer commented:
I added the config and fixed the black thing. Black itself has currently an import issue in the pre-commit and pylint complains locally, so I forced git to commit anyway and am interested in how it runs here.

Poolitzer commented:
I found the issue with black psf/black#2964, we either need to pin the click dependency to a lower version or upgrade the black version, what would you prefer and should I squeeze it in here or in a second PR?

harshil21 commented:
@Poolitzer We&#x27;re updating dependencies in <a href="https://github.com/python-telegram-bot/python-telegram-bot/pull/2925">#2925</a> so imo no need to do it here

<a href="https://github.com/python-telegram-bot/python-telegram-bot/pull/2925">PR #2925: Update Test Dependencies versions by harshil21</a>

Bibo-Joshi commented:
@harshil21 I&#x27;m okay with doing it here/in asyncio as the CI currently fails otherwise. #2925 is due only after asyncio

Poolitzer commented:
@Bibo-Joshi So to be less disruptive, I could hardpin the dependency, and we remove the hardpin and upgrade the version in #2925

Bibo-Joshi reviewed (approved):
The changes themself LGTM 👍 did you by any chance already test the examples? if not: do you want to before we merge or do we just do that later?

Poolitzer commented:
@Bibo-Joshi I didn&#x27;t test it, I think that makes most sense before the first release.

harshil21 commented on examples/inlinebot.py:
```diff
@@ -56,14 +56,14 @@ async def inlinequery(update: Update, context: CallbackContext.DEFAULT_TYPE) ->
             title="Italic",
             input_message_content=InputTextMessageContent(
-                f"_{escape_markdown(query)}_", parse_mode=ParseMode.MARKDOWN
+                f"_{escape(query)}_", parse_mode=ParseMode.MARKDOWN
```
```suggestion
                f"{escape(query)}", parse_mode=ParseMode.HTML
```

Poolitzer replied:
wupps, good catch

harshil21 commented:
Only thing I didn't run was passport, payment, and chatmemberbot. Rest all are working fine! Asyncio runs smooooth!

Bibo-Joshi commented on examples/rawapibot.py:
```diff
@@ -10,51 +10,56 @@
 
 
-async def echo(bot: telegram.Bot) -> None:
+async def echo(bot: Bot, update_id: int) -> None:
```
```suggestion
async def echo(bot: Bot, update_id: int) -> int:
```

harshil21 replied:
hm weird mypy didn't catch that

Bibo-Joshi commented on examples/rawapibot.py:
```diff
@@ -10,51 +10,56 @@
             # Reply to the message
+            logger.info("Found message %s!", update.message.text)
             await update.message.reply_text(update.message.text)
+        return next_update_id
```
```suggestion
        return next_update_id
    return update_id
```

Poolitzer commented:
looks good to me

Bibo-Joshi commented:
Tested chatmemberbot. Runs fine except for a but in `CMU.difference` that I found (see #2947), but that's not due to the example.

Bibo-Joshi commented:
Tested passport & payment example. For passports, I adjusted the html file a bit and found #253 - which makes me want to stabilize `Updater` a bit more, but that's not for this PR.

If CI runs through & @harshil21 and @Poolitzer don't have anything to add, we can merge :)

harshil21 commented:
All good from my side

Poolitzer commented:
Looks good to me as well
